ansible/roles/debops.postfix_virtual_ldap/defaults/main.yml

```diff
@@ -0,0 +1,685 @@
+            else "absent" }}'
+
+  # Postfix is final destination for the specified list of domains; mail is delivered
+  # via the $virtual_transport mail delivery transport
+  - name: 'virtual_mailbox_domains'
+    value: 'ldap:/etc/postfix/ldap_virtual_domains.cf'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  # Optional lookup tables that alias specific mail addresses or domains to other local or remote address.
+  - name: 'virtual_alias_maps'
+    value: '{{ postfix_virtual_ldap__virtual_alias_maps | join(",") }}'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  #A prefix that the virtual delivery agent prepends to all pathname results from $virtual_mailbox_maps table lookups.
+  # This is a safety measure to ensure that an out of control map doesn't litter the file system with mailboxes.
+  - name: 'virtual_mailbox_base'
+    value: '{{ postfix_virtual_ldap__mailbox_base }}'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  # Optional lookup tables with all valid addresses in the domains that match $virtual_mailbox_domains.
+  - name: 'virtual_mailbox_maps'
+    value: 'ldap:/etc/postfix/ldap_virtual_recipients.cf'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  # Lookup tables with the per-recipient user ID that the virtual delivery agent uses while writing
+  # to the recipient's mailbox.
+  - name: 'virtual_uid_maps'
+    value: 'static:{{ postfix_virtual_ldap__vmail_posix_uidnumber }}'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  # Lookup tables with the per-recipient group ID for virtual mailbox delivery.
+  - name: 'virtual_gid_maps'
+    value: 'static:{{ postfix_virtual_ldap__vmail_posix_gidnumber }}'
+    section: 'virtual'
+    state: '{{ "present"
+            if (postfix_virtual_ldap__ldap_enabled|bool and
+                postfix__version is version_compare("2.0", ">="))
+            else "absent" }}'
+
+  # The minimum user ID value that the virtual delivery agent accepts as a result from $virtual_uid_maps table lookup
+  - name: 'virtual_minimum_uid'
+    value: '{{ postfix_virtual_ldap__vmail_posix_uidnumber }}'
+    section: 'virtual'
+    state: 'absent'
```

Comment:
This is currently a workaround. I have problem injecting the correct int value to the main.cf config. Somehow I get an OID value, where it should be an INT. Really weird.

Reply:
It might be not an OID but an IPv4 address from a weird subnet, like `1.0.4.3` or something. Can you share the configuration you are trying to apply? I can then check it on my end and see what happens.

Reply:
When the `virtual_minimum_uid` option is set to `state: present` I get the following result:

```
# --------------------- Virtual mail configuration ---------------------

virtual_mailbox_domains = ldap:/etc/postfix/ldap_virtual_domains.cf
virtual_alias_maps = $alias_maps,ldap:/etc/postfix/ldap_virtual_aliases.cf
virtual_mailbox_base = /var/vmail
virtual_mailbox_maps = ldap:/etc/postfix/ldap_virtual_recipients.cf
virtual_uid_maps = static:997
virtual_gid_maps = static:992
virtual_minimum_uid = 0.0.3.229    #<<--- WTF!?
```

The string there should be `997`, but I am unable to get this value correctly. Filtering `postfix_virtual_ldap__vmail_posix_uidnumber | int` to an INTEGER does not help either.

Reply:
Nevertheless this config option can be actually removed:

- We have only one UID & GID storing the emails to `/var/vmail`, this is the `vmail` posix system user
- Current postfix virtual mail config does not allows to "virtual users" to access the mailDir directly, because the MailDir is not in the home folder of these users and the users may not be POSIX users but just Virtual Mail Users.
- So, without direct access to the filesystem (e.g. over SSH), users cannot read the MailDir directly, thus no need to specify different UIDs/GIDs.
- Mail reading will be handled entirely by Dovecot over IMAP. And dovecot will access the MailDir folder as the `vmail` user.

I hope it is clear....